fmlloader/build.gradle

```
@@ -72,16 +71,36 @@ tasks.withType(JavaCompile) {
 
 license {
     header = rootProject.file('LICENSE-header.txt')
+}
 
-    include 'net/minecraftforge/'
+publishing {
+    publications.register('mavenJava', MavenPublication).configure {
+        from components.java
+        artifactId = 'fmlloader'
+        pom {
+            name = project.name
+            description = 'Modifactions to Minecraft to enable mod developers.'
+            url = 'https://github.com/MinecraftForge/MinecraftForge'
+            PomUtils.setGitHubDetails(pom, 'MinecraftForge')
+            license PomUtils.Licenses.LGPLv2_1
+        }
+    }
+    
+    repositories {
+        maven gradleutils.publishingForgeMaven
+    }
 }
 
-artifacts {
-    archives jar
-    archives sourcesJar
+tasks.register('writeForgeVersionJson') {
+    doLast {
+        file('src/main/resources/forge_version.json').json = [
+            'forge': FORGE_VERSION,
+            'mc': MC_VERSION,
+            'mcp': MCP_VERSION
+        ] as LinkedHashMap
+    }
 }
 
-publishing.publications.mavenJava(MavenPublication) {
-    from components.java
-    artifact sourcesJar
-}
\ No newline at end of file
+tasks.named('generateResources').configure {
+    dependsOn(writeForgeVersionJson)
```

I think this causes eager configuration of the 'writeForgeVersionJson' task - it should probably be
```suggestion
    dependsOn 'writeForgeVersionJson'
```
instead

---

It shouldnt because its in the config stage. but meh